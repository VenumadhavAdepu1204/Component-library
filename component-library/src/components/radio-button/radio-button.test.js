import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import RadioButton from './index'

describe('<RadioButton />', () => {
  const { container } = render(<RadioButton onChange={jest.fn()} />)

  test('renders properly', () => {
    expect(container).toMatchSnapshot()
  })

  test('handleKeyPress function is called', () => {
    const { getAllByDisplayValue } = render(<RadioButton onChange={jest.fn()} value='radio button' name='test' />)
    fireEvent.keyPress(getAllByDisplayValue('radio button')[0], {
      key: 'Enter',
      charCode: 13
    })
  })

  test('handleKeyPress function is else called', () => {
    const { getAllByDisplayValue } = render(<RadioButton onChange={jest.fn()} value='radio button' name='test' />)
    fireEvent.keyPress(getAllByDisplayValue('radio button')[0], {
      key: 'Tab'
    })
  })
})
