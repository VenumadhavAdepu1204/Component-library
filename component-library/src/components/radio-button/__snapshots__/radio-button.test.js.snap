// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButton /> renders properly 1`] = `
<div>
  <div>
    <input
      class="sc-bdVaJa hKBQmx"
      data-test-id="undefined-radio-button"
      tabindex="0"
      type="radio"
      value=""
    />
    <label
      class="sc-bwzfXH ikeyHx"
    />
  </div>
</div>
`;
